src/nomad_simulations/schema_packages/numerical_settings.py

```diff
@@ -890,15 +911,295 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
         super().normalize(archive, logger)
 
 
-class BasisSet(NumericalSettings):
-    """"""
-
-    # TODO work on this base section (@ndaelman-hu)
+class BasisSet(ArchiveSection):
+    """A type section denoting a basis set component of a simulation.
+    Should be used as a base section for more specialized sections."""
 
     def __init__(self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs):
         super().__init__(m_def, m_context, **kwargs)
         # Set the name of the section
         self.name = self.m_def.name
 
+
+class PlaneWaveBasisSet(BasisSet, Mesh):
+    """
+    Defines a plane-wave basis set over a singular mesh.
+    """
+
+    cutoff_energy = Quantity(
+        type=np.float64,
+        unit='joule',
+        description="""
+        Cutoff energy for the plane-wave basis set. The simulation uses plane waves with energies below this cutoff.
+        """,
+    )
+
+    smearing_method = Quantity(
+        type=MEnum('Fermi-Dirac', 'Gaussian', 'Methfessel-Paxton'),
+        description="""
+        Method used to smear the occupation numbers in the plane-wave basis set.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class LocalizedBasisSet(BasisSet):
+    """
+    Section that connects a basis set to a localized unit, often an atom.
+    """
+
+    species_definition = SubSection(sub_section=AtomsState.m_def)
+    # ? @JosePizarro3: are we ensure that the atoms_state is always set? can you point me to any normalizer?
```

Comment:
Applied. Tbh, at first I was somewhat hesitant to use `AtomsState`. It contains all the relevant information, but I'm unsure in how far it's supposed to represent an actual "physical" system. Since Wannier uses it too, I guess other basis sets should be fine.